src/components/SearchBar/SearchBar.js

```diff
@@ -1,15 +1,34 @@
+import React from "react";
+import { withRouter } from "react-router-dom";
 import { StyledInput } from "./searchBar.styles";
 
-const onSubmitHandler = (e) => {
-  e.preventDefault();
-};
+class SearchBar extends React.Component {
+  state = {
+    value: "",
+  };
 
-const SearchBar = () => {
-  return (
-    <form onSubmit={onSubmitHandler}>
-      <StyledInput placeholder="Search" />
-    </form>
-  );
-};
+  handleChange = (e) => {
+    const { value } = e.target;
+    this.setState({ value });
+  };
```

Comment:
This looks fancy but I cant understand how can it work... wouldnt it be better to do this.setState({value: e.target.value}) straight away?

Reply:
this is 50/50

 `handleChange = ({target: {value}}) => {
    this.setState({ value });
  };`

 `handleChange = ({target}) => {
    this.setState({ value: target.value });
  };`

 `handleChange = (e) => {
    this.setState({ value: e.target.value });
  };`

these are all viable options, because this is such a concise function, doesn't really matter. Context is everything and it's about 'do I get what this function is about?'